src/host/utf8handling.c

```diff
@@ -0,0 +1,33 @@
+/**
+ * \file   utf8handking.c
+ * \brief  Handles conversions between Unicode (UTF-8) and system native encoding (wide chars on Windows)
+ * \author Copyright (c) 2017 Jérôme Leclercq and the Premake project
+ */
+
+#include "premake.h"
+#include "stdlib.h"
+
+#ifdef PLATFORM_WINDOWS
+const char* utf8_fromwide(lua_State* L, const wchar_t* wstr)
```

Comment:
Mmh, actually a wrapper function is useless if all it does is calling `WideCharToMultiByte`, since we can do that in places where we are calling this function (since we're already checking for the Windows part).

What if I make something like that:
```c
const char* utf8_fromwide(const wchar_t* wstr)
 {
 	int size_required = WideCharToMultiByte(CP_UTF8, 0, wstr, -1, NULL, 0, NULL, NULL);
 
 	char* unicode_str = (char*) malloc(size_required * sizeof(char));
 	WideCharToMultiByte(CP_UTF8, 0, wstr, -1, unicode_str, size_required, NULL, NULL);
 
 	return unicode_str;
 }
```

be a better solution? Of course the caller has to free the buffer then (which I dislike).